tests/test_symbolics.py

```diff
@@ -298,7 +298,10 @@ def test_extended_sympy_arithmetic():
 def test_integer_abs():
     i1 = Dimension(name="i1")
     assert ccode(Abs(i1 - 1)) == "abs(i1 - 1)"
-    assert ccode(Abs(i1 - .5)) == "fabs(i1 - 5.0e-1F)"
+    assert ccode(Abs(i1 - .5)) == "fabsf(i1 - 5.0e-1F)"
+    assert ccode(
+        Abs(i1 - Constant('half', dtype=np.float64, default_value=0.5))
+    ) == "fabs(i1 - half)"
```

Comment:
I'm not happy with this, but don't 100% know how I'm supposed to generate a literal double precision `0.5` symbolically

Reply:
Just np.float64(0.5)

Reply:
```python
>>> ccode(Abs(i1 - np.float64(0.5)))
'fabs(i1 - 5.0e-1F)'
```
It looks like the float64 is being demoted to float32, this could point to an issue elsewhere